build-tests/api-extractor-scenarios/etc/test-outputs/ambientNameConflict/api-extractor-scenarios.api.json

```diff
@@ -18,6 +19,7 @@
         {
           "kind": "Function",
           "canonicalReference": "(ambientNameConflict:0)",
+          "uid": "api-extractor-scenarios/ambientNameConflict(Promise`1{void},api-extractor-scenarios/Promise`1{void})",
```

Comment:
Here you are referring to `Promise`, which is an ambient type picked up from the global context, which may be different for each invocation of the compiler.  In this case it's probably the compiler's built-in type, but e.g. it could also be the `React` or `Jest` global variable.  Or in some files it could be an explicitly imported symbol that occludes the global variable.

It seems imprecise for the UID to depend on definitions that are not part of the API surface.

Also, I'd be curious to see what the output would be for a complex type like https://github.com/microsoft/web-build-tools/pull/1331#discussion_r293485337, since people really do write code like that in practice.  

And what the corresponding URL would be, given that URLs need to be case-insensitive.

It might be more efficient to chat about this offline, but as desirable as stable UIDs are, I'm really not convinced that it's appropriate for TypeScript.